ood-portal-generator/lib/ood_portal_generator/dex.rb

```diff
@@ -64,12 +61,29 @@ def self.installed?
       File.directory?(config_dir) && File.executable?('/usr/sbin/ondemand-dex')
     end
 
+    def self.default_auth
+      ['AuthType openid-connect', 'Require valid-user'].freeze
+    end
+
     def self.config_dir
       '/etc/ood/dex'
     end
 
     private
 
+    # determine if this config would enable dex configurations.
+    def set_enable(config)
+      @enable = if config.nil?
+                  false
+                elsif config.respond_to?(:empty?) && config.empty?
+                  false
+                elsif config == false
+                  false
+                else
+                  true
```

Comment:
We should raise an Error here if we're about to return true and `!Dex.installed?` but I'd like to defer that if we could.

Reply:
I think it's fine to defer